scene/resources/visual_shader.cpp

```diff
@@ -1762,6 +1762,9 @@ void VisualShaderNodeGroupBase::set_inputs(const String &p_inputs) {
 	for (int i = 0; i < input_port_count; i++) {
 
 		Vector<String> arr = input_strings[i].split(",");
+		ERR_FAIL_INDEX(0, arr.size());
```

Comment:
I think, there should be check for exactly 3 arguments instead (ERR_FAIL_COND(arr.size() != 3))

Reply:
Actually, I did that on purpose. If I checked in _!= 3_ way, we cannot understand exactly which index of the error is in the editor. But only because I check the index, I can see that the error is exactly in _nth_ index.

Reply:
Ah, nvm then

Reply:
`ERR_FAIL_INDEX` is meant to compare a user-passed index with an array size, not the other way around.
This should simply be `ERR_FAIL_COND(arr.size() != 3);` as suggested.

Users won't get a lot more information from `Index 2=2 out of size (arr.size()=3)` than a plain `Condition ' arr.size() != 3 ' is true.` IMO. The latter is more explicit about the fact that something should have a size of 3.